### PR TITLE
 439, Contributor Discovery Endpoint added. 

### PR DESCRIPTION
**Pull request by NathanHelm**
# NOTE TO TA (explanation): this PR is slightly oversized because I had to write extensive tests + edge case testing because I was adding a new system. I couldn't resize my contributor function because it would break logic.
## 📝 Description

- Added /local-llm/context/contributors endpoint for the local LLM workflow.
- User can now find user contributions given a project ID (relative path) from /context endpoint.  

See screenshots for more evidence

**Closes:** #439

---

## 🔧 Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ New feature (non-breaking change that adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation added/updated
- [x] ✅ Test added/updated
- [ ] ♻️ Refactoring
- [ ] ⚡ Performance improvement

---

## 🧪 Testing

> Please describe how you tested this PR (both manually and with tests).  
> Provide instructions so we can reproduce.

- test_contributors_endpoint_exists - Endpoint is registered
- test_discover_contributors_valid_selection - Valid repo selection succeeds
- test_discover_contributors_missing_active_context - Returns 404 without active intake
- test_discover_contributors_invalid_repo_ids - Returns 422 for invalid repo IDs
- test_discover_contributors_mixed_valid_invalid - Returns 422 for mixed valid/invalid IDs
- test_discover_contributors_missing_repo_ids - Returns 422 when repo_ids missing
- test_discover_contributors_empty_repo_ids - Returns 422 for empty repo_ids list
- test_discover_contributors_response_contract - Response matches schema contract
- test_discover_contributors_multiple_repos - Discovers contributors across multiple repos
- test_discover_contributors_subset_selection - Can select subset of available repos
- test_contributors_endpoint_in_openapi - Endpoint appears in OpenAPI schema

---

## ✓ Checklist

- [ ] 🤖 GenAI was used in generating the code and I have performed a self-review of my own code
- [ ] 💬 I have commented my code where needed
- [ ] 📖 I have made corresponding changes to the documentation
- [ ] ⚠️ My changes generate no new warnings
- [ ] ✅ I have added tests that prove my fix is effective or that my feature works and tests are passing locally
- [ ] 🔗 Any dependent changes have been merged and published in downstream modules
- [ ] 📱 Any UI changes have been checked to work on desktop, tablet, and/or mobile

---

## 📸 Screenshots

- here I'm using shlok's mock project: 

1) I'm calling context and getting ids
Screenshot 2026-03-15 at 1 52 33 PM
2) I'm calling context/contributors with path id. It's returning the contributor (shlok)
Screenshot 2026-03-15 at 1 52 55 PM

### CODE REVIEW

File: src/artifactminer/api/local_llm.py [hunk 1]
```diff
@@ -143,6 +173,107 @@ def _discover_repos_in_zip(zip_path: str) -> List[RepositoryCandidate]:
...
+                
+                # Update name if we get a better one
+                if name and (not contrib["name"] or len(name) > len(contrib["name"])):
+                    contrib["name"] = name
+        
+        except subprocess.TimeoutExpired:
+            # Timeout is an error - repo may be corrupted or too large
+            raise ValueError(f"Git operation timed out for {repo_path}")
+        except Exception as e:
+            # Other errors (e.g., permission issues) are warnings - skip repo
```
Comment (van-cpu) [inline src/artifactminer/api/local_llm.py]: This block here continues silently on errors, we might want to have it log the error or raise something.
